arangod/Aql/Functions.cpp

```diff
@@ -2409,6 +2409,46 @@ AqlValue Functions::Document(arangodb::aql::Query* query,
   return AqlValue(AqlValueHintNull());
 }
 
+/// @brief function MATCHES
+AqlValue Functions::Matches(arangodb::aql::Query* query,
+                             transaction::Methods* trx,
+                             VPackFunctionParameters const& parameters) {
+  ValidateParameters(parameters, "MATCHES", 2, 3);
+
+  AqlValue docToFind = ExtractFunctionParameterValue(trx, parameters, 0);
+  AqlValue exampleDocs = ExtractFunctionParameterValue(trx, parameters, 1);
+
+  bool retIdx = false;
+  if (parameters.size() == 3) {
+    retIdx = ExtractFunctionParameterValue(trx, parameters, 2).toBoolean();
+  }
+
+  AqlValueMaterializer materializer(trx);
+  VPackSlice docSlice = materializer.slice(docToFind, false);
+
+  int32_t idx = -1;
+  for (auto const& name : VPackArrayIterator(materializer.slice(exampleDocs, false))) {
+    idx++;
+    if (basics::VelocyPackHelper::VPackEqual()(name, docSlice)) {
```

Comment:
better use VelocyPackHelper::compare(...) with Options being passed. 
```
VPackOptions* options = trx->transactionContextPtr()->getVPackOptions ...
```
If no Options are passed, then documents cannot be compared.